### PR TITLE
Improve packet state machine

### LINKED ISSUE #1678: java.lang.NegativeArraySizeException Crash at SdlPsm.java line 241 com.smartdevicelink.transport.SdlPsm.transitionOnInput
### Bug Report
Fatal Crash 
```
Fatal Exception: java.lang.NegativeArraySizeException: -592969471
       at com.smartdevicelink.transport.SdlPsm.transitionOnInput(SdlPsm.java:241)
       at com.smartdevicelink.transport.SdlPsm.handleByte(SdlPsm.java:88)
       at com.smartdevicelink.transport.MultiplexBluetoothTransport$ConnectedThread.run(MultiplexBluetoothTransport.java:801)

```

##### Reproduction Steps
Seeing crash in Firebase Crashlytics dash, unfortunately no reproduction steps to share. 

##### Expected Behavior
No Crash

##### Observed Behavior
Crashing

##### OS & Version Information
* Android Version: 46% Android 11, 45% Android 10, 5% Android 7%
* SDL Android Version: 4.12.0
* Testing Against: 
According to crash logs. app was always in background when the crash was seen.

##### Test Case, Sample Code, and / or Example App
[Paste a link to a PR, gist, or other code that exemplifies this behavior]

### PR DESCRIPTION
**Pull request by joeygrover**
Fixes #1678 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have run the unit tests with this PR
- [x] I have tested this PR against an IVI system
- [x] I have tested Android, Java SE, and Java EE

#### Unit Tests
Additional tests were added to test the SdlPsm.java class and some tests were refactored. These included more corner case issues as provided in #1678. 

#### Core Tests
Basic smoke testing can be done to ensure the packet state machine still works as intended. It is difficult to perform tests that would have actually broken the PSM so we rely on the unit tests for those scenarios.

### Summary
- Surrounded the base operation of the SdlPsm class in a try/catch that will set the state to ERROR state and safely exit. This prevents exceptions from being thrown in developer's applications. A log is written out through the DebugTool
- Added a check for correct service types. This helps with recovery tremendously. 
- Added better dataSize checking per early versions of the protocol. Dynamic MTUs are not included, but adding them might not actually help that much since they could be extremely large anyways.
- Refactored some checks to reduce redundant logic.
 
##### Enhancements
* Added better checks into the SdlPsm class
 
##### Bug Fixes
* Added safeguards to prevent crashes from propagating to developer's apps.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: base/src/main/java/com/smartdevicelink/transport/SdlPsm.java [hunk 1]
```diff
@@ -32,15 +32,19 @@
 package com.smartdevicelink.transport;
 
 import com.smartdevicelink.protocol.SdlPacket;
+import com.smartdevicelink.protocol.enums.SessionType;
```
Comment (JulianKast) [inline base/src/main/java/com/smartdevicelink/transport/SdlPsm.java:35]: This added import is unused
```suggestion
```

**Review by JulianKast — CHANGES_REQUESTED** [review]

**Review by JulianKast — APPROVED** [review]
